### DISCUSSION TITLE
Support for prefixing console with reverse proxy path

### DISCUSSION THREAD

**Discussion by jdheyburn** [minio/minio#15551] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

I'm looking to see if its possible to have my minio instance behind a reverse proxy, both the server and the console, at the same domain.

I'd like to have my server available at `https://minio.example.com`, and the console at `https://minio.example.com/ui/`.

I'm aware that minio supports path-style requests such that `https://minio.example.com/bucket/object` can be used to retrieve `object` in `bucket`, and so this would conflict with a bucket named `ui`. However seeing as buckets cannot be fewer than 3 characters, this conflict will not happen.

I'm using Caddy to reverse proxy, where my config is:

```Caddy
[minio.example.com](http://minio.example.com/) {
  reverse_proxy /ui/* localhost:CONSOLE_PORT
  reverse_proxy localhost:SERVER_PORT
}
```

This is causing some stuff to break on the console, see the screenshot below. I am unsure if it because the web server doesn't know that its assets are available at `/ui/` and not `/`.

![image](https://user-images.githubusercontent.com/34041368/185458233-d8def669-1b90-4310-9345-af9647e83250.png)

**Describe the solution you'd like**

A configurable way to define the root path that the console is being served at, behind the reverse proxy.

**Describe alternatives you've considered**

I _could_ just have another virtual host for the web interface (i.e. `minio-ui.example.com`, but I like the simplicity of having a single domain.

**Additional context**

Please let me know if there's a better way of doing the above - thank you!

**Comment by harshavardhana** [comment, marked as answer]
You need to just say `MINIO_BROWSER_REDIRECT_URL=http://minio.domain.com/console`  @jdheyburn 

and add `location /console { }` point to your proxy_pass to relevant port

**Reply by jdheyburn** [reply]
@harshavardhana  Awesome thanks! Apologies for missing that in the config.

Is my assumption correct that setting this won't impact server API functionality? I see you used `console` as an example, would that then mean you could not use `console` as a bucket name?

**Reply by harshavardhana** [reply]
> Is my assumption correct that setting this won't impact server API functionality? I see you used console as an example, would that then mean you could not use console as a bucket name?

Yes you cannot use `console` as bucketname to avoid that can use reserved bucket name `minio` instead.

**Comment by emouawad** [comment]
@harshavardhana Is there an equivalent helm chart value? using https://github.com/minio/minio/tree/master/helm/minio with Istio VirtualService - A destination to minio-console is showing the same as the above screenshot - related to https://github.com/minio/console/issues/2149 and https://github.com/minio/console/issues/1908

**Reply by suhailsid** [reply]
Hello team is there any way I am running minio on Linux server , I am using minio latest version but the issue is current version comes with embeded console , 9000 port automatic redirect to 9001 , and all my S3 connectivity establish on 443 as minio running with revers proxy , how can I access minio console and minio API on 443 using same revers proxy to overcome this issue

**Comment by suhailsid** [comment]
> location /consol

@harshavardhana 
will this solution work in latest minio running on Linux server, I tried but console browser showing three image on left and not the correct UI.

**Comment by anbraten** [comment]
My caddy-v2 config for this:

```
s3.my-domain.tld {
	handle_path /console* {
		reverse_proxy minio:9001
	}

	reverse_proxy minio:9000
}
```

Somehow I have to open `https://s3.my-domain.tld/console/login` to be able to login. Everything else seems to work fine.

**Reply by rallisf1** [reply]
> Somehow I have to open `https://s3.my-domain.tld/console/login` to be able to login. Everything else seems to work fine.

I know I'm late to the party, but just add `redir /console /console/` as the first directive
